### ISSUE TITLE
[Feature Request] Keeping Keras Callback functionality in Estimators created from `keras.model_to_estimator`

### ISSUE THREAD

**Issue by zmjjmz** [tensorflow/tensorflow#21979]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: n/a
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: n/a
- **TensorFlow installed from (source or binary)**: n/a
- **TensorFlow version (use command below)**: 1.9
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: n/a
- **GCC/Compiler version (if compiling from source)**: n/a
- **CUDA/cuDNN version**: n/a
- **GPU model and memory**: n/a
- **Exact command to reproduce**: n/a

### Describe the problem

Currently if I want to define a model in Keras and train it in a distributed fashion, the easiest/most-apparent method is to convert it into an Estimator using [model_to_estimator](https://www.tensorflow.org/api_docs/python/tf/keras/estimator/model_to_estimator). Unfortunately this has a few issues, and one of them is that any [Keras Callbacks](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback) go unpreserved.

Theoretically these callbacks could be converted into some form of [`SessionRunHook`](https://www.tensorflow.org/api_docs/python/tf/train/SessionRunHook) although the SessionRunHook doesn't know things that the Callback does (e.g. epoch information). However, this does not happen at all -- the `EstimatorSpec` that's returned out of the `model_fn` from `_create_keras_model_fn` [does not specify any sort of `training_hooks`](https://github.com/tensorflow/tensorflow/blob/r1.9/tensorflow/python/estimator/keras.py#L400) regardless of whether or not the Keras model has Callbacks defined.

I'm sure this is a non-trivial problem to solve, however it is frustrating that `model_to_estimator` causes a loss in functionality like this.

Are there any plans to remedy this going forward?

Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Jasonnor** [comment]
Any progress on it? 🚀

**Comment by beasteers** [comment]
Wait yeah this is kinda big... I wish I knew that before spending a couple hours trying to decipher tf.estimator docs and outdated examples getting the rest of it set up.

**Comment by albenoit** [comment]
Hi, any news about this or another way to do the same thing ?

**Comment by lucaskawazoi** [comment]
Hi, any news about this or another way to do the same thing ?

**Comment by chrishmorris** [comment]
This is an important issue. With a Keras model, you can reduce the learning rate on plateau, then stop training if the loss still does not improve. Powerful, and avoids a lot of experiment with LR schedules.

**Comment by rmothukuru** [comment]
@zmjjmz,
Please refer [this tutorial](https://www.tensorflow.org/tutorials/distribute/keras) which shows how to use **`Keras Callbacks`**, `TensorBoard`, `ModelCheckpoint`, `LearningRateScheduler` etc.. and to `Train` the `Keras Model` in `Distributed Fashion`.

Please refer [this Guide](https://www.tensorflow.org/guide/distributed_training) for more information on `Training` a `Tensorflow Keras Model` in `Distributed Fashion`.

Hope this helps. Thanks!

**Comment by zmjjmz** [comment]
Hey there,

Glad to see that 2.5 years later, this is now possible :) I'll close this issue.
